### PR TITLE
JTBD Guide Review for Read application data

### PR DESCRIPTION
**Pull request by heatheramz**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx [hunk 1]
```diff
@@ -2,3 +2,572 @@ export const meta = {
   title: `Read application data`,
   description: `Learn more about how to read data using Amplify's GraphQL API category`
 };
+
```
Comment (heatheramz) [inline src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx:5]: @renebrandel This one only has the new content under JS and React Native inline filter for the guide. The rest of the platforms use their existing fragments without any guide language or blending.

File: src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx [hunk 2]
```diff
@@ -2,3 +2,572 @@ export const meta = {
...
+<Callout>
+
+__Note:__ The cost structure of your underlying data source can impact the cost to run some queries. For example, the `list` operation uses DynamoDB "scan operations" which uses more read request units than the `get` operation. You will want to review the associated costs for these operations for your data source. In our example case here we are using Amazon DynamoDB and you can [learn more about how DynamoDB pricing is calculated](https://aws.amazon.com/dynamodb/pricing/) on their site.
+
+</Callout>
+
+The Amplify CLI's codegen automatically creates code for all possible GraphQL operations—mutations, queries, and subscriptions. For JavaScript applications, this generated code is saved in the src/graphql folder.
```
Comment (heatheramz) [inline src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx:30]: @renebrandel Line 30 TODO Another codegen section we had marked to revisit

File: src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx [hunk 3]
```diff
@@ -2,3 +2,572 @@ export const meta = {
...
+</Block>
+</BlockSwitcher>
+
+</Accordion>
+
+## Paginate your data
+  
+As your data grows, you will want to filter and paginate your list queries. Fortunately, this is already built in to the Amplify-generated GraphQL API.
+
+You can find the input schemas in the Docs pane of the GraphQL explorer or inside your autogenerated `/graphql` folder. They look like this:
```
Comment (heatheramz) [inline src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx:162]: @renebrandel Line 162 TODO Follow up on note "the content below here might be a bit too technical as the starting point. (and also duplicative with paginate section lower down.)"

**Review by heatheramz — COMMENTED** [review]
@renebrandel Updated the read application data guide to review on its own and transferred unresolved questions from the previous PR.

File: src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx [hunk 4]
```diff
@@ -2,3 +2,572 @@ export const meta = {
...
+  query: listTodos, variables: variables
+});
+
+const { items: itemsPage2 } = res.data.listTodos;
+```
+
+</Block>
+</BlockSwitcher>
+
+<Accordion title='Implement pagination in your GraphQL API' headingLevel='3' eyebrow='Walkthrough'>
```
Comment (renebrandel) [inline src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx:350]: @heatheramz - it's worth considering to remove this accordion. Rereading it, I feel like it's just a more expanded version of the code snippet above without additional value/detail.

Comment (heatheramz) [reply src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx:360]: @renebrandel  We can - this is from the guide - https://docs.amplify.aws/guides/api-graphql/graphql-pagination/q/platform/js/ which is an area we are looking to merge content or deprecate as needed. If the updated guidance here provides the same then the recommendation will be to retire the JS guide (and other platforms as this page is expanded).  Is there anything from the accordion that we would want to add into the main page before it is deleted?

Comment (heatheramz) [reply src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx:360]: Discuss 9/26 - keeping for now - add to UX if possible

File: src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx [hunk 5]
```diff
@@ -2,3 +2,582 @@ export const meta = {
...
+}
+```
+
+These filters vary based on the type of the field, but are linked to corresponding [Amazon DynamoDB queries](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Condition.html).
+
+</Accordion>
+
+### Compound filters
+
+You can combine filters with `and`, `or`, and `not` Boolean logic. Observe, in the autogenerated schema above, that `ModelTodoFilterInput` is recursive in respect to those fields. So if, for example, you wanted to filter for `priority` values of 1 OR 2, you would do this:
```
Comment (heatheramz) [inline src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx:242]: Just noticed the "ModelTodoFilterInput' reference here is about text that is in the accordion. Suggest we change it to direct them to that accordion vs saying "above".

File: src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx [hunk 6]
```diff
@@ -2,3 +2,604 @@ export const meta = {
...
 };
+
+<InlineFilter filters={["js", "react-native"]}>
+
+You can read application data using the Amplify GraphQL client. In this guide we will review the difference between reading data and getting data, how to filter query results to get just the data you need, and how to paginate results to make your data more manageable. We will also show you how to cancel these requests when needed.
+
+Before you begin, you will need:
+
+<InlineFilter filters={["js"]}>
+
```
Comment (heatheramz) [inline src/pages/lib/graphqlapi/query-data/q/platform/[platform].mdx:13]: Noting here again that js is linking to the new page so will want to consider as we push this to main.

**Pull request merged**